android-components/components/support/webextensions/src/main/java/mozilla/components/support/webextensions/WebExtensionSupport.kt

```diff
@@ -234,7 +235,28 @@ object WebExtensionSupport {
                     if (!extension.isBuiltIn()) {
                         store.dispatch(
                             WebExtensionAction.UpdatePromptRequestWebExtensionAction(
-                                WebExtensionPromptRequest.PostInstallation(extension),
+                                WebExtensionPromptRequest.AfterInstallation.PostInstallation(extension),
+                            ),
+                        )
+                    }
+                }
+
+                override fun onInstallationFailedRequest(
+                    extension: WebExtension?,
+                    exception: WebExtensionInstallException,
+                ) {
+                    if (extension == null) {
+                        store.dispatch(
+                            WebExtensionAction.UpdatePromptRequestWebExtensionAction(
+                                WebExtensionPromptRequest.BeforeInstallation.InstallationFailed(null, exception),
+                            ),
+                        )
+                        return
+                    }
+                    if (!extension.isBuiltIn()) {
```

Comment:
is this actually happening? like, could this method be called for a built-in extension?

Reply:
I'm not completely sure, just added the check just in case as it will be really odd for an users. Will try to simulated an installation error one of our built-in to see.

Reply:
Verified with [dummy-incompatible.xpi](https://searchfox.org/mozilla-central/source/mobile/android/geckoview/src/androidTest/assets/web_extensions/dummy-incompatible.xpi) an xip we host on m-c. built-in extensions have a different error handling that other extension, we can remove the check as this code path is not executed :)